modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java

```diff
@@ -239,11 +239,19 @@ private void _addDataDefinitionFieldLinks(
 			long fieldSetDDMStructureId = GetterUtil.getLong(
 				ddmFormField.getProperty("ddmStructureId"));
 
-			if (fieldSetDDMStructureId != 0) {
+			DEDataDefinitionFieldLink deDataDefinitionFieldLink =
 				_deDataDefinitionFieldLinkLocalService.
-					addDEDataDefinitionFieldLink(
-						groupId, classNameId, dataDefinitionId,
-						fieldSetDDMStructureId, ddmFormField.getName());
+					fetchDEDataDefinitionFieldLinks(
+						classNameId, dataDefinitionId, fieldSetDDMStructureId,
+						ddmFormField.getName());
+
+			if (fieldSetDDMStructureId != 0) {
+				if (deDataDefinitionFieldLink == null) {
+					_deDataDefinitionFieldLinkLocalService.
+						addDEDataDefinitionFieldLink(
+							groupId, classNameId, dataDefinitionId,
+							fieldSetDDMStructureId, ddmFormField.getName());
+				}
 
 				_addDataDefinitionFieldLinks(
```

Comment:
@manuelRives can we simplify this code to:

```
long fieldSetDDMStructureId = GetterUtil.getLong(
	ddmFormField.getProperty("ddmStructureId"));

if (fieldSetDDMStructureId == 0) {
	continue;
}

DEDataDefinitionFieldLink deDataDefinitionFieldLink =
	_deDataDefinitionFieldLinkLocalService.
		fetchDEDataDefinitionFieldLinks(
			classNameId, dataDefinitionId, fieldSetDDMStructureId,
			ddmFormField.getName());

if (deDataDefinitionFieldLink == null) {
	_deDataDefinitionFieldLinkLocalService.
		addDEDataDefinitionFieldLink(
			groupId, classNameId, dataDefinitionId,
			fieldSetDDMStructureId, ddmFormField.getName());
}

_addDataDefinitionFieldLinks(
	classNameId, dataDefinitionId,
	ddmFormField.getNestedDDMFormFields(), groupId);
```
? 

This way we don't need to `fetchDEDataDefinitionFieldLinks` if `fieldSetDDMStructureId == 0`.